libs/pika/async_mpi/include/pika/async_mpi/mpi_helpers.hpp

```diff
@@ -175,16 +133,47 @@ namespace pika::mpi::experimental::detail {
                 else
                 {
                     // pass the result onto a new task and invoke the continuation
-                    auto snd0 = ex::just(status) | ex::transfer(default_pool_scheduler(p)) |
-                        ex::then([receiver = PIKA_MOVE(receiver)](int status) mutable {
-                            PIKA_DETAIL_DP(
-                                mpi_tran<5>, debug(str<>("set_value_error_helper"), status));
-                            set_value_error_helper(status, PIKA_MOVE(receiver));
+                    auto snd0 = ex::transfer_just(default_pool_scheduler(p)) |
+                        ex::then([receiver = PIKA_MOVE(receiver)]() mutable {
+                            PIKA_DETAIL_DP(mpi_tran<5>, debug(str<>("set_value")));
+                            ex::set_value(PIKA_FORWARD(Receiver, receiver));
                         });
                     ex::start_detached(PIKA_MOVE(snd0));
                 }
             },
             request);
     }
 
+    // -----------------------------------------------------------------
+    // handler_method::continuation
+    // adds a request callback to the mpi polling code which will call
+    // the set_value/set_error helper using the void return signature
+    template <typename OperationState>
+    void add_continuation_request_callback(MPI_Request request, OperationState& op_state)
+    {
+        detail::add_request_callback(
+            [&op_state](int status) mutable {
+                PIKA_DETAIL_DP(mpi_tran<5>, debug(str<>("callback_void")));
+                set_value_error_helper(status, PIKA_MOVE(op_state.receiver));
+            },
+            request);
+    }
+
+    // -----------------------------------------------------------------
+    // handler_method::mpix_continuation
+    /// typedef int (MPIX_Continue_cb_function)(int rc, void *cb_data);
+    template <typename OperationState>
+    static int mpix_callback([[maybe_unused]] int rc, void* cb_data)
```

Comment:
Can or should we do something with `rc`? I assume it's an error code, so if it's not "success" we should likely propagate that to receivers?